src/Cms/Responder.php

```diff
@@ -284,6 +292,15 @@ public function headers(array|null $headers = null)
 			// inject CORS headers if enabled
 			$corsHeaders = Cors::headers();
 			if ($corsHeaders !== []) {
+				foreach ($corsHeaders as $name => $value) {
+					if ($name === 'Vary') {
+						$corsVaryValues = array_map('trim', explode(',', $value));
+						$this->markRequestDependentHeader('Vary', $corsVaryValues);
+					} else {
+						$this->markRequestDependentHeader($name);
+					}
+				}
```

Comment:
`Thought`: This causes side effects in a getter by changing the volatile header list whenever the headers are retrieved. This could lead to unexpected behavior and is not considered clean code.

`Suggestion`: We could instead dynamically inject the CORS headers into the `volatileHeaders` list in the `cacheHeaders()`/`toCacheArray()` method just like we dynamically inject them into the header list here. Then both getters do their thing without influencing each other. Disadvantage would be that the CORS headers need to be collected twice. Maybe you know a way to avoid that, but I would say the performance impact should still be pretty low unless the user defined a custom `cors` callback option that does expensive processing.